enos/modules/test_e2e/main.tf

```diff
@@ -0,0 +1,62 @@
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+variable "test_package" {
+  description = "Name of Go test package to run"
+  type        = string
+}
+variable "alb_boundary_api_addr" {
+  description = "URL of the Boundary instance"
+  type        = string
+}
+variable "auth_method_id" {
+  description = "Id of Auth Method used to login to Boundary instance"
+  type        = string
+}
+variable "auth_login_name" {
+  description = "Name of admin user"
+  type        = string
+}
+variable "auth_password" {
+  description = "Password of admin user"
+  type        = string
+}
+variable "local_boundary_dir" {
+  description = "Local Path to boundary executable"
+  type        = string
+}
+variable "aws_ssh_private_key_path" {
+  description = "Local Path to key used to SSH onto created hosts"
+  type        = string
+}
+variable "target_ips" {
+  description = "List of IP Addresses of created hosts"
+  type        = list(string)
+}
+
+locals {
+  aws_ssh_private_key_path = abspath(var.aws_ssh_private_key_path)
+}
+
+resource "enos_local_exec" "run_e2e_test" {
+  environment = {
+    BOUNDARY_ADDR                             = var.alb_boundary_api_addr,
+    BOUNDARY_AUTHMETHOD_ID                    = var.auth_method_id,
+    BOUNDARY_AUTHENTICATE_PASSWORD_LOGIN_NAME = var.auth_login_name,
+    BOUNDARY_AUTHENTICATE_PASSWORD_PASSWORD   = var.auth_password,
+    BOUNDARY_E2E_TARGET_IP                    = var.target_ips[0],
+    BOUNDARY_E2E_SSH_USER                     = "ubuntu"
+    BOUNDARY_E2E_SSH_KEY_PATH                 = local.aws_ssh_private_key_path,
+  }
+
+  inline = ["PATH=\"${var.local_boundary_dir}:$PATH\" go test -v ${var.test_package}"]
```

Comment:
Good question... I believe it currently uses whatever is on your machine.

Reply:
Not sure if this answers your question, but after talking with the enos team, their plan is to use this github action: https://github.com/actions/setup-go and supply `go.mod` to determine the version. 

Example in vault: https://github.com/hashicorp/vault-enterprise/blob/main/.github/workflows/build.yml

Reply:
I mostly asked the question to make you think about it 😁. Specifying the version via the action is fine. Note that setting the Go version in `go.mod` is only used to specify the _minimum_ supported Go version in the module, it doesn't manage the Go installation for you. I expect in reality this will just be updated together with all the other version strings when we update our Go version.